### PR TITLE
9.0 auth admin passkey

### PR DESCRIPTION
**Pull request by rvalyi**
same as 8.0 https://github.com/OCA/server-tools/pull/519
but also migrated the view and enable the module on 9.0
cc @legalsylvain @pedrobaeza

### CODE REVIEW

**Comment by pedrobaeza** [comment]
:+1:

**Comment by pedrobaeza** [comment]
Please change module version to 9.0.1.0.0.

**Comment by pedrobaeza** [comment]
See the comments in 8.0 version.

File: auth_admin_passkey/__openerp__.py [hunk 1]
```diff
@@ -41,6 +24,6 @@
     'images': [],
     'post_load': '',
```
Comment (lasley) [inline auth_admin_passkey/__openerp__.py:25]: Please remove empty keys

File: auth_admin_passkey/model/res_config.py [hunk 2]
```diff
@@ -1,76 +1,53 @@
...
-        ),
-    }
+    @api.multi
+    def set_auth_admin_passkey_send_to_user(self):
+        for config in self:
+            self.env['ir.config_parameter'].set_param(
+                "auth_admin_passkey.send_to_user",
+                config.auth_admin_passkey_send_to_user or '')
+
+    auth_admin_passkey_send_to_admin = fields.Boolean(
```
Comment (lasley) [inline auth_admin_passkey/model/res_config.py:43]: Please move column definitions above the methods

File: auth_admin_passkey/__openerp__.py [hunk 3]
```diff
@@ -1,24 +1,7 @@
...
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# -*- coding: utf-8 -*-
+# Copyright (C) 2013-2014 GRAP (http://www.grap.coop)
+# @author Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
 
 {
     'name': 'Authentification - Admin Passkey',
```
Comment (JayVora-SerpentCS) [inline auth_admin_passkey/__openerp__.py:7]: Should not it be 'Authentication' ?

File: auth_admin_passkey/__init__.py [hunk 4]
```diff
@@ -1,23 +1,6 @@
...
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# -*- coding: utf-8 -*-
+# Copyright (C) 2013-2014 GRAP (http://www.grap.coop)
+# @author Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
```
Comment (JayVora-SerpentCS) [inline auth_admin_passkey/__init__.py:5]: May be Add Copyright section for Akretion!

File: auth_admin_passkey/model/res_users.py [hunk 5]
```diff
@@ -1,55 +1,40 @@
...
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
 
 import datetime
 
 from openerp import SUPERUSER_ID
-from openerp import pooler
+from openerp import registry
 from openerp import exceptions
-from openerp.osv.orm import Model
+from openerp import models, api
```
Comment (lasley) [inline auth_admin_passkey/model/res_users.py:11]: IMO it is inconsistent to perform all the other `openerp` imports on one line, then do two here. Please either move the `api` import into its own line or, preferably, group them in one line.

File: auth_admin_passkey/model/res_users.py [hunk 6]
```diff
@@ -1,55 +1,40 @@
...
-        icp_obj = self.pool['ir.config_parameter']
-        admin_user = self.browse(cr, SUPERUSER_ID, SUPERUSER_ID)
-        login_user = self.browse(cr, SUPERUSER_ID, user_id)
-        send_to_admin = safe_eval(icp_obj.get_param(
-            cr, SUPERUSER_ID, 'auth_admin_passkey.send_to_admin', 'True'))
-        send_to_user = safe_eval(icp_obj.get_param(
-            cr, SUPERUSER_ID, 'auth_admin_passkey.send_to_user', 'True'))
+        mail_obj = self.env['mail.mail']
+        icp_obj = self.env['ir.config_parameter']
+        admin_user = self.sudo().browse(SUPERUSER_ID)
```
Comment (lasley) [inline auth_admin_passkey/model/res_users.py:30]: Wouldn't `self.sudo().env.user` work?

File: auth_admin_passkey/model/res_users.py [hunk 7]
```diff
@@ -1,55 +1,40 @@
...
+        login_user = self.sudo().browse(self.env.uid)
```
Comment (lasley) [inline auth_admin_passkey/model/res_users.py:31]: Why not `self.env.user.sudo()`?

File: auth_admin_passkey/model/res_users.py [hunk 8]
```diff
@@ -58,34 +43,34 @@ def _send_email_passkey(self, cr, user_id, user_agent_env):
...
-                cr, mail['lang'], _('Passkey used'))
+                mail['lang'], _('Passkey used'))
             body = self._get_translation(
-                cr, mail['lang'],
+                mail['lang'],
                 _("""Admin user used his passkey to login with '%s'.\n\n"""
                     """\n\nTechnicals informations belows : \n\n"""
                     """- Login date : %s\n\n""")) % (
                         login_user.login,
                         datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S"))
```
Comment (lasley) [inline auth_admin_passkey/model/res_users.py:53]: `fields.Datetime.now()`

File: auth_admin_passkey/model/res_users.py [hunk 9]
```diff
@@ -58,34 +43,34 @@ def _send_email_passkey(self, cr, user_id, user_agent_env):
...
 
-    def _send_email_same_password(self, cr, login_user):
+    @api.cr
+    def _send_email_same_password(self, login_user):
         """ Send a email to the admin user to inform that another user has the
  same password as him."""
-        mail_obj = self.pool['mail.mail']
-        admin_user = self.browse(cr, SUPERUSER_ID, SUPERUSER_ID)
+        mail_obj = self.env['mail.mail']
+        admin_user = self.sudo().browse(SUPERUSER_ID)
```
Comment (lasley) [inline auth_admin_passkey/model/res_users.py:66]: `self.sudo().env.user`

File: auth_admin_passkey/model/res_users.py [hunk 10]
```diff
@@ -94,20 +79,20 @@ def _send_email_same_password(self, cr, login_user):
     def authenticate(self, db, login, password, user_agent_env):
         """ Authenticate the user 'login' is password is ok or if
  is admin password. In the second case, send mail to user and admin."""
-        user_id = super(res_users, self).authenticate(
+        user_id = super(ResUsers, self).authenticate(
             db, login, password, user_agent_env)
         if user_id and (user_id != SUPERUSER_ID):
             same_password = False
-            cr = pooler.get_db(db).cursor()
+            cr = registry(db).cursor()
```
Comment (lasley) [inline auth_admin_passkey/model/res_users.py:86]: I think `self.pool.cursor()` is the way to get the cursor. I don't see any examples of this way in v9 code

**Comment by rvalyi** [comment]
@lasley @JayVora-SerpentCS thanks for the review. Guys I'll apply these changes on https://github.com/OCA/server-tools/pull/519 for the remaining ones and then cherry-pick and amend the commit from 8.0 here. cc @legalsylvain . Once I started making changes in the 8.0 PR I put the "work in progress" label on this 9.0 PR, I'm sorry if that wasn't clear.

File: auth_admin_passkey/__init__.py (hunk 4 above)
Comment (rvalyi) [reply auth_admin_passkey/__init__.py:5]: Well I'll push a new feature after this API refactor and v9 migration and then I'll mention myself as a contributor indeed.

**Comment by pedrobaeza** [comment]
I have merged the other PR. Please include the changes and attend comments here.

**Comment by JonathanNEMRY** [comment]
I think this branche should be "rebase" to be complient with the last commit 601109eef017ac2c07290ad23b81cd92f0b70b48 . Moreover, what's the current status of this one?

File: auth_admin_passkey/model/res_users.py (hunk 5 above)
Comment (JonathanNEMRY) [reply auth_admin_passkey/model/res_users.py:11]: hi @lasley  I think it is OK for this one cause models/api always are on the same line but other can be accepted into multiple line instruction. I am ok to let it as it is. what do you think about it?

File: auth_admin_passkey/model/res_users.py (hunk 7 above)
Comment (JonathanNEMRY) [reply auth_admin_passkey/model/res_users.py:31]: @lasley  I think that `self.sudo().browse(uid)` is different than `self.sudo().browse(uid)` actually the first one return a `recordset` which contains an `environement` generated with user `uid` and the current one return a record set of the user.

File: auth_admin_passkey/model/res_users.py (hunk 6 above)
Comment (JonathanNEMRY) [reply auth_admin_passkey/model/res_users.py:30]: same comment here

File: auth_admin_passkey/model/res_users.py (hunk 9 above)
Comment (JonathanNEMRY) [reply auth_admin_passkey/model/res_users.py:66]: same here

File: auth_admin_passkey/model/res_users.py (hunk 10 above)
Comment (JonathanNEMRY) [reply auth_admin_passkey/model/res_users.py:86]: This way to return the cursor is possible into 9.0 source code as Odoo do it into its mail addons :smile:

**Comment by JonathanNEMRY** [comment]
I finally applied changes into https://github.com/akretion/server-tools/pull/14 & https://github.com/akretion/server-tools/pull/15 Once @rvalyi merged 15 and 14 I think this one will be ok to be merged into oca/9.0

**Comment by pedrobaeza** [comment]
@rvalyi this one is conflicting. Are you going to finish this?

**Pull request closed without merging**